code/math/curve.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) Volition, Inc. 1999.  All rights reserved.
+ *
+ * All source code herein is the property of Volition, Inc. You may not sell 
+ * or otherwise commercially exploit the source or things you created based on the 
+ * source.
+ *
+*/ 
+
+#include "math/curve.h"
+#include "parse/parselo.h"
+
+SCP_vector<Curve> Curves;
+
+int curve_get_by_name(SCP_string in_name) {
+	for (int i = 0; i < (int)Curves.size(); i++) {
+		if (Curves[i].name == in_name)
+			return i;
+	}
+
+	return -1;
+}
+
+void parse_curve_table(const char* filename) {
+	try
+	{
+		read_file_text(filename, CF_TYPE_TABLES);
+		reset_parse();
+
+		optional_string("#Curves");
+
+		while (optional_string("$Name:")) {
+			SCP_string name;
+			stuff_string(name, F_NAME);
+
+			int index = curve_get_by_name(name);
+
+			if (index < 0) {
+				Curve curv = Curve(name);
+				curv.ParseData();
+				Curves.push_back(curv);
+			} else {
+				Curves[index].ParseData();
+			}
+		}
+	}
```

Comment:
Probably want a `required_string("#End")` here

Reply:
Ehh, I'm not a fan of overly strict table syntax, unless there's a good reason to I wasn't planning on bothering.

Reply:
It's fine for now.  The main reason is if another section were ever added after it.  IIRC one of the existing tables didn't do that and it made things trickier.